docs/topics/downloader-middleware.rst

```diff
@@ -496,6 +496,59 @@ In order to use this storage backend:
 .. _LevelDB: https://github.com/google/leveldb
 .. _leveldb python bindings: https://pypi.python.org/pypi/leveldb
 
+.. _httpcache-storage-custom:
+
+Implementing custom cache storage backend
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can implement custom cache storage backend by creating a Python class that
+defines the methods described below.
+
+.. module:: scrapy.extensions.httpcache
+
+.. class:: CacheStorage
+
+    .. method:: open_spider(spider)
+
+      This method gets called after a spider has been opened for crawling.
```

Comment:
You should probably mention that this method handles the `open_spider` signal, and link to the documentation of that signal using the special `:signal:` syntax, e.g. ``` :signal:`spider_opened` ```. Same for `close_spider` below.